opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -294,7 +297,8 @@ namespace Opm
                 {
                     cellToConnectionIdx_[cell_idx_[idx]] = idx;
 
-                    auto cellFacesRange = cell2Faces[cell_idx_.at(idx)];
+                    const int cell_index = cartesian_to_compressed_.at(cell_idx_[idx]);
```

Comment:
Should we try to provide some good message here to see this cell is not active, we need to fix the deck or we discard this cell? Maybe for now, we `throw` to terminate the simulation until we found it is okay to discard some inactive cells?

Reply:
I just hit this case running a model on my local workstation and the simulation terminated because the exception was not caught.  I've made the following change locally, but I don't know that it's the correct solution.  Any suggestion for what the real solution is?
```diff
diff --git a/opm/autodiff/AquiferFetkovich.hpp b/opm/autodiff/AquiferFetkovich.hpp
index 02bde94b..c9f36795 100755
--- a/opm/autodiff/AquiferFetkovich.hpp
+++ b/opm/autodiff/AquiferFetkovich.hpp
@@ -267,7 +267,12 @@ namespace Opm
             cellToConnectionIdx_.resize(ebos_simulator_.gridView().size(/*codim=*/0), -1);
             for (size_t idx = 0; idx < cell_idx_.size(); ++idx)
             {
-              const int cell_index = cartesian_to_compressed_.at(cell_idx_[idx]);
+              auto c = this->cartesian_to_compressed_.find(cell_idx_[idx]);
+              if (c == this->cartesian_to_compressed_.end()) {
+                // No such cell.  Skip.
+                continue;
+              }
+              const int cell_index = c->second;
               cellToConnectionIdx_[cell_index] = idx;
               const auto cellFacesRange = cell2Faces[cell_index];
               for(auto cellFaceIter = cellFacesRange.begin(); cellFaceIter != cellFacesRange.end(); ++cellFaceIter)
```

Reply:
Without the ECL results, I guess we should give a warning message, while we need to remove that connection. There might be some quantities associated with that connection, we need to do some cleaning up and synchronization. 

Or, we should do some checking/cleaning up even before this to remove the connections that located in in-active cells.

Reply:
Or, these connections in the in-active cells should not enter the model class at all.  They can stay in the parser (even be removed from the parser), while some pre-processing can be done before the Aquifer** classes are instantiated.